Makefile

```diff
@@ -207,6 +207,19 @@ generate: controller-gen
 e2e: chainsaw
 	$(CHAINSAW) test --test-dir ./tests/e2e
 
+# end-to-end-test for testing automatic RBAC creation
+.PHONY: e2e-automatic-rbac
+e2e-automatic-rbac: chainsaw
+	# Add extra needed permissions
+	kubectl apply -f hack/rbac-operator-permissions.yaml
+	kubectl rollout restart deployment opentelemetry-operator-controller-manager -n opentelemetry-operator-system
+	kubectl rollout status deployment opentelemetry-operator-controller-manager -n opentelemetry-operator-system
+	go run hack/check-operator-ready.go
+	$(CHAINSAW) test --test-dir ./tests/e2e-automatic-rbac
+	# Cleanup
+	kubectl delete -f hack/rbac-operator-permissions.yaml
+	kubectl rollout restart deployment opentelemetry-operator-controller-manager -n opentelemetry-operator-system
```

Comment:
I think I'd prefer if this was done the same way feature gates are enabled in e2e tests - by adding a patch to the kustomization. I think you can just add in the permissions you need that way, without needing to redeclare the whole ClusterRole, which is brittle.

As a general rule, when we do custom setup in e2e tests, we should do it the same way as we'd imagine users applying the same customization in their clusters.

Reply:
Done